vtysh/vtysh.c

```diff
@@ -2878,11 +2878,16 @@ static void backup_config_file(const char *fbackup)
 
 	/* Move current configuration file to backup config file. */
 	if (unlink(integrate_sav) != 0) {
-		vty_out(vty, "Warning: %s unlink failed\n", integrate_sav);
+		if (errno != ENOENT) {
+			vty_out(vty, "Unlink failed for %s: %s\n",
+				integrate_sav, strerror(errno));
+		}
 	}
 	if (rename(fbackup, integrate_sav) != 0) {
-		vty_out(vty, "Error renaming %s to %s\n", fbackup,
-			integrate_sav);
+		if (errno != ENOENT) {
```

Comment:
is this check necessary?

>       ENOENT The link named by oldpath does not exist; or, a direc‐
>              tory  component in newpath does not exist; or, oldpath
>              or newpath is an empty string.

Reply:
This check allows us to bypass the warning message when frr.conf is missing.

The unlink() call will see ENOENT if /etc/frr/frr.conf.sav doesn't exist, while this rename() call will see ENOENT if /etc/frr/frr.conf doesn't exist.

If the user is getting to this point then they are already trying to write their runtime config. If they were missing their frr.conf file before the write, then personally I don't see much use in calling that out now.

Reply:
Yeah, I got the idea of the patch, I read it wrong and thought the first case covered this one. This one covers the case where neither `frr.conf` or `frr.conf.sav` exist, and we try to rename `frr.conf` to `frr.conf.sav`, if I read it correctly.

Reply:
Consolidate tests into single if statement, as per your offline suggestion. Thanks!